lib/puppet/provider/iam_user/v2.rb

```diff
@@ -5,9 +5,29 @@
 
   mk_resource_methods
 
+  def self.get_users
+    user_results = iam_client.list_users()
+    users = user_results.users
+
+    truncated = user_results.is_truncated
+    marker = user_results.marker
+
+    while truncated and marker
+      Puppet.debug('iam_user results truncated, proceeding with discovery')
+      response = iam_client.list_users({marker: marker})
+      response.users.each {|u|
```

Comment:
multiline do/end is also possible, but should only be used if you do multiple statements in it, like this:

```ruby
response.users.each do user
  users << user
  puts user
end
```